api.go

```diff
@@ -1002,6 +1007,114 @@ func marshalResponse(resp interface{}, w http.ResponseWriter, status int, r *htt
 	return nil
 }
 
+func filterSparseFields(resp interface{}, r *http.Request) (interface{}, error) {
+	query := r.URL.Query()
+	queryParams := parseQueryFields(&query)
```

Comment:
It only checks if there are query parameters. But yes the regex compiles every time in the parseQueryFields function. I can let it compile once that would be better.